features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/ActiveCallManager.kt

```diff
@@ -111,6 +130,12 @@ class DefaultActiveCallManager @Inject constructor(
             delay(ElementCallConfig.RINGING_CALL_DURATION_SECONDS.seconds)
             incomingCallTimedOut(displayMissedCallNotification = true)
         }
+
+        // Acquire a wake lock to keep the device awake during the incoming call, so we can process the room info data
+        if (activeWakeLock?.isHeld == true) {
+            activeWakeLock.release()
+        }
+        activeWakeLock?.acquire(ElementCallConfig.RINGING_CALL_DURATION_SECONDS * 1000L)
```

Comment:
Can we use the non timeout version here and keep the one from before?
Also, it doesn't seem to be thread safe, not sure on which thread those methods are called

Reply:
Actually, if I remove it I get this lint warning:

> Provide a timeout when requesting a wakelock with PowerManager. Wakelock. acquire(long timeout). This will ensure the OS will cleanup any wakelocks that last longer than you intend, and will save your user's battery

Which I think makes sense, and it's safer with the timeout.

About the thread safety, do we want to just wrap every function with a mutex lock?

Reply:
If we have suspending method everywhere, yes I guess

Reply:
Actually I tried that and the result is not great, it leads to adding new dependencies, makes the tests more complex... 🫤

But I'm not sure if using `@Synchronized` or a synchronized lock would be a good solution either.

Reply:
In the end I used `Mutex`. Let me know if it's better.